### PR TITLE
    [FLINK-30566] Add benchmark for AgglomerativeClustering, HashingTF, IDF, KBinsDiscretizer, LinearRegression, LinearSVC, LogisticRegression, NGram, RegexTokenizer, Tokenizer and Vectorindexer

### PR DESCRIPTION
**Pull request by zhipeng93**
## What is the purpose of the change
- Add benchmark configurations for some of the existing algorithms.

## Brief change log
- Add benchmark configuration files for AgglomerativeClustering, HashingTF, IDF, KBinsDiscretizer, LinearRegression, LinearSVC, LogisticRegression, NGram, RegexTokenizer, Tokenizer and Vectorindexer

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-ml-benchmark/src/main/resources/linearregression-benchmark.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "version": 1,
+  "linearregression": {
+    "stage": {
+      "className": "org.apache.flink.ml.regression.linearregression.LinearRegression",
+      "paramMap": {
+        "featuresCol": "features",
```
Comment (lindong28) [inline flink-ml-benchmark/src/main/resources/linearregression-benchmark.json:22]: Would it be simpler (and more consistent with existing benchmark configs) to skip specifying the default values for `featuresCol`?

Same for `labelCol`, `weightCol`, `inputCol` and `outputCol`.

**Review by lindong28 — COMMENTED** [review]
Thanks for the PR. LGTM overall. Left one minor comment.

File: flink-ml-benchmark/src/main/resources/linearregression-benchmark.json (hunk 1 above)
Comment (zhipeng93) [reply flink-ml-benchmark/src/main/resources/linearregression-benchmark.json:22]: I hold a bit different opinion here.

Given that we have to specify column names for input data `features, labels, weight`, adding this configurations in the algorithms (i.e., `linear regression` here) is supposed to make it easier to follow. What do you think?

Comment (lindong28) [reply flink-ml-benchmark/src/main/resources/linearregression-benchmark.json:22]: Strictly speaking, if a user expects to use the default value for a column parameter, he/she would not have to specify the column name, right?

Could you come up with a use-case for explicitly specifying the default column name for those column parameters? It is typically useful to specify values for parameters that users are likely to tune (e.g. input data count). IMO it is unlikely users would care the column names used in a benchmark. Thus it seems simpler to skip them. What do you think?

Comment (zhipeng93) [reply flink-ml-benchmark/src/main/resources/linearregression-benchmark.json]: Thanks for the detailed explanation. I agree that skipping the default column names are more friendly to users.

**Comment by zhipeng93** [comment]
Thanks for the review @lindong28 . I have updated the PR in the latest PR. Please take a look.

**Comment by lindong28** [comment]
Thanks for the update! LGTM.

**Pull request merged**